contracts/chain-adapters/interfaces/ArbitrumForwarderInterface.sol

```diff
@@ -0,0 +1,285 @@
+     * @param maxFeePerGas price bid for L3 execution. Should not be set to 1 (magic value used to trigger the RetryableData error)
+     * @param tokenTotalFeeAmount amount of fees to be deposited in native token to cover for retryable ticket cost
+     * @param data ABI encoded data of L3 message
+     * @return unique message number of the retryable transaction
+     */
+    function createRetryableTicket(
+        address to,
+        uint256 l3CallValue,
+        uint256 maxSubmissionCost,
+        address excessFeeRefundAddress,
+        address callValueRefundAddress,
+        uint256 gasLimit,
+        uint256 maxFeePerGas,
+        uint256 tokenTotalFeeAmount,
+        bytes calldata data
+    ) external returns (uint256);
+}
+
+/**
+ * @notice Generic gateway contract for bridging standard ERC20s to Arbitrum-like networks.
+ */
+interface ArbitrumERC20GatewayLike {
+    /**
+     * @notice Deposit ERC20 token from Ethereum into Arbitrum-like networks.
+     * @dev L3 address alias will not be applied to the following types of addresses on L2:
+     *      - an externally-owned account
+     *      - a contract in construction
+     *      - an address where a contract will be created
+     *      - an address where a contract lived, but was destroyed
+     * @param _l2Token L2 address of ERC20
+     * @param _refundTo Account, or its L3 alias if it have code in L2, to be credited with excess gas refund in L3
+     * @param _to Account to be credited with the tokens in the L3 (can be the user's L3 account or a contract),
+     * not subject to L3 aliasing. This account, or its L3 alias if it have code in L2, will also be able to
+     * cancel the retryable ticket and receive callvalue refund
+     * @param _amount Token Amount
+     * @param _maxGas Max gas deducted from user's L3 balance to cover L3 execution
+     * @param _gasPriceBid Gas price for L3 execution
+     * @param _data encoded data from router and user
+     * @return res abi encoded inbox sequence number
+     */
+    function outboundTransferCustomRefund(
+        address _l2Token,
+        address _refundTo,
+        address _to,
+        uint256 _amount,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        bytes calldata _data
+    ) external payable returns (bytes memory);
+
+    /**
+     * @notice get ERC20 gateway for token.
+     * @param _token ERC20 address.
+     * @return address of ERC20 gateway.
+     */
+    function getGateway(address _token) external view returns (address);
+}
+
+/**
+ * @notice Contract containing logic to send messages from L2 to Arbitrum-like L3s.
+ * @dev This contract is meant to share code for Arbitrum L2 forwarder contracts deployed to various
+ * different L2 architectures (e.g. Base, Arbitrum, ZkSync, etc.). It assumes that the L3 conforms
+ * to an Arbitrum-like interface.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+abstract contract ArbitrumForwarderInterface {
```

Comment:
nit: `Interface` typically implies that no functions are implemented.

`ArbitrumForwarderBase` or just `ArbitrumForwarder` might be better.

Reply:
I like `ArbitrumForwarderBase` https://github.com/across-protocol/contracts/pull/609/commits/0de93033f590bd7886fae198bf05dc4b36e9732e